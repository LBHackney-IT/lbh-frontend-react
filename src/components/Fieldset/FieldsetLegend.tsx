import classNames from "classnames";
import PropTypes from "prop-types";
import React from "react";

import { Attributes, DataAttributes } from "../../helpers/Attributes";

import "lbh-frontend/lbh/components/lbh-fieldset/_fieldset.scss";

/**
 * The proptypes for {@link FieldsetLegend}.
 *
 * This also supports all `aria-*` and `data-*` attributes.
 *
 * @noInheritDoc
 */
export interface FieldsetLegendProps
  extends React.AriaAttributes,
    DataAttributes {
  id?: string;
  className?: string;
  /**
   * Whether the legend also acts as the heading for the page. Places the legend
   * children inside a `h1` if set to `true`.
   */
  isPageHeading?: boolean;
  children: React.ReactNode;
}

/**
 * A component to set a legend for a {@link Fieldset} by passing
 * in via the {@link FieldsetProps.legend} prop.
 *
 * ```ts
 * // Creating a fieldset with a legend
 * <Fieldset
 *   id="fieldsetId"
 *   className="fieldsetClass"
 *   describedBy="fieldsetDescribedBy"
 *   role="fieldsetRole"
 *   legend={
 *     <FieldsetLegend id="legendId" className="legendClass" isPageHeading>
 *       children
 *     </FieldsetLegend>
 *   }
 * >
 *   children
 * </Fieldset>
 * ```
 */
export const FieldsetLegend: React.FunctionComponent<FieldsetLegendProps> = (
  props: FieldsetLegendProps
): JSX.Element => {
  const { id, className, isPageHeading, children } = props;

  const extraAttributes = Attributes.ariaAndData(props);

  return (
    <legend
      id={id}
      className={classNames(
        "govuk-fieldset__legend",
        "lbh-fieldset__legend",
        className
      )}
      {...extraAttributes}
    >
      {isPageHeading ? (
        <h1 className="govuk-fieldset__heading lbh-fieldset__heading">
          {children}
        </h1>
      ) : (
        children
      )}
    </legend>
  );
};

FieldsetLegend.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  isPageHeading: PropTypes.bool,
  children: PropTypes.node.isRequired
};
