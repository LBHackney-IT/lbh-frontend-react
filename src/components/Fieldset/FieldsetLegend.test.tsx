import React from "react";
import { create } from "react-test-renderer";

import { FieldsetLegend } from "./FieldsetLegend";

it("renders correctly with all props", () => {
  const component = create(
    <FieldsetLegend
      id="Testid"
      className="class1 class2"
      isPageHeading
      aria-label="test fieldset legend"
      data-test="test data"
    >
      Test
    </FieldsetLegend>
  );

  expect(component).toMatchInlineSnapshot(`
    <legend
      aria-label="test fieldset legend"
      className="govuk-fieldset__legend lbh-fieldset__legend class1 class2"
      data-test="test data"
      id="Testid"
    >
      <h1
        className="govuk-fieldset__heading lbh-fieldset__heading"
      >
        Test
      </h1>
    </legend>
  `);
});

it("renders correctly without optional props", () => {
  const component = create(<FieldsetLegend>Test</FieldsetLegend>);

  expect(component).toMatchInlineSnapshot(`
    <legend
      className="govuk-fieldset__legend lbh-fieldset__legend"
    >
      Test
    </legend>
  `);
});
