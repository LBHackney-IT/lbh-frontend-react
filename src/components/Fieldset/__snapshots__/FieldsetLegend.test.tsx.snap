// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<legend
  className="govuk-fieldset__legend lbh-fieldset__legend class1 class2"
  id="Testid"
>
  <h1
    className="govuk-fieldset__heading lbh-fieldset__heading"
  >
    Test
  </h1>
</legend>
`;

exports[`renders correctly without optional props 1`] = `
<legend
  className="govuk-fieldset__legend lbh-fieldset__legend"
>
  Test
</legend>
`;
